internal/processing/istio/authorization_policy_processor.go

```diff
@@ -0,0 +1,173 @@
+package istio
+
+import (
+	"context"
+	"fmt"
+
+	gatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"github.com/kyma-incubator/api-gateway/internal/builders"
+	"github.com/kyma-incubator/api-gateway/internal/processing"
+	"istio.io/api/security/v1beta1"
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AuthorizationPolicyProcessor is the generic processor that handles the Istio Authorization Policies in the reconciliation of API Rule.
+type AuthorizationPolicyProcessor struct {
+	Creator AuthorizationPolicyCreator
+}
+
+// AuthorizationPolicyCreator provides the creation of Authorization Policies using the configuration in the given APIRule.
+// The key of the map is expected to be unique and comparable with the
+type AuthorizationPolicyCreator interface {
```

Comment:
This interface seems not to be needed, as in all likelihood
 there will be only one implementation for Istio handler